app/models/piece.rb

```diff
@@ -27,6 +27,22 @@ def off_board?(new_x, new_y)
     new_x < 1 || new_x > 8 || new_y < 1 || new_y > 8
   end
 
+  def move_to!(new_x, new_y)
+    @moving_piece = self.piece
```

Comment:
We can ask @ilyakrasnov about this, but I'm not sure we need to use self methods on lines 31-33.  I think a possible refactoring might be:

`moving_piece = Piece.find(params[:id])`

Reply:
@Nikkovichert why do you want to use `moving_piece`? you can just continue referring to self.. like self.x_position and keep operating on self..

Reply:
@PaleoGuy2012 what you suggest here is only possible in the controller (where we have access to params) - but in general its not necessary to find a piece - the piece is `self`

Reply:
seems like we keep checking if x and y are nil? Maybe make a method                         def nil_move?(new_x, new_y)
    new_x == x_position && new_y == y_position 
  end                        and clean up     def move_to!